GPU/Common/TextureReplacer.cpp

```diff
@@ -256,31 +257,57 @@ bool TextureReplacer::LoadIniValues(IniFile &ini, bool isOverride) {
 				ERROR_LOG(G3D, "Unsupported syntax under [hashes]: %s", item.first.c_str());
 			}
 		}
+	}
 
-		// Now, translate the filenameMap to the final aliasMap.
-		for (auto &pair : filenameMap) {
-			std::string alias;
-			int mipIndex = 0;
-			for (auto &level : pair.second) {
-				if (level.first == mipIndex) {
-					alias += level.second + "|";
-					mipIndex++;
-				} else {
-					WARN_LOG(G3D, "Non-sequential mip index %d, breaking. filenames=%s", level.first, level.second.c_str());
-					break;
-				}
+	// Scan the root of the texture folder/zip and preinitialize the hash map.
+	std::vector<File::FileInfo> filesInRoot;
+	dir->GetFileListing("/", &filesInRoot, nullptr);
+	for (auto file : filesInRoot) {
+		if (file.isDirectory)
+			continue;
+		if (file.name.empty() || file.name[0] == '.')
+			continue;
+		Path path(file.name);
+		std::string ext = path.GetFileExtension();
+
+		std::string hash = file.name.substr(0, file.name.size() - ext.size());
+		if (!((hash.size() >= 26 && hash.size() <= 27 && hash[24] == '_') || hash.size() == 24)) {
+			continue;
+		}
+		// OK, it's hash-like enough to try to parse it into the map.
+		if (equalsNoCase(ext, ".ktx2") || equalsNoCase(ext, ".png") || equalsNoCase(ext, ".dds")) {
```

Comment:
Well, I guess we might as well add ".zim"?

-[Unknown]

Reply:
late night coding, oops.